src/context.rs

```diff
@@ -255,22 +255,29 @@ impl<'a> Context<'a> {
     }
 
     /// Will lazily get repo root and branch when a module requests it.
-    pub fn get_repo(&self) -> Result<&Repo, git2::Error> {
-        self.repo.get_or_try_init(|| -> Result<Repo, git2::Error> {
-            let repository = if env::var("GIT_DIR").is_ok() {
-                Repository::open_from_env()
-            } else {
-                let dirs: [PathBuf; 0] = [];
-                Repository::open_ext(&self.current_dir, git2::RepositoryOpenFlags::FROM_ENV, dirs)
-            }?;
-            Ok(Repo {
-                branch: get_current_branch(&repository),
-                workdir: repository.workdir().map(Path::to_path_buf),
-                path: Path::to_path_buf(repository.path()),
-                state: repository.state(),
-                remote: get_remote_repository_info(&repository),
+    pub fn get_repo(&self) -> Result<&Repo, git_repository::discover::Error> {
+        self.repo
+            .get_or_try_init(|| -> Result<Repo, git_repository::discover::Error> {
+                let shared_repo =
+                    ThreadSafeRepository::discover_with_environment_overrides(&self.current_dir)?;
+
+                let repository = shared_repo.to_thread_local();
+                let branch = get_current_branch(&repository);
+                let remote = get_remote_repository_info(&repository, branch.as_deref());
+                let path = repository.path().to_path_buf();
+                let trust_level = git_sec::Trust::from_path_ownership(&path)
```

Comment:
Ideally `git-repository` users won't ever have the need to pull in a plumbing crate directly. Thus `git_sec` should be `git::sec` here unless it's a matter of preference.

The reason for that is that they will only have to maintain one crate version to use anything `git` related, which is `git-repository`. Note that as per [gitoxide's stability policy](https://github.com/Byron/gitoxide/blob/498072ea42dca7b9d00bedba42829bdac92195b9/STABILITY.md#L1-L0) breaking changes in exposed dependencies of `git-repository` will automatically lead to a breaking semver change in `git-repository` transitively, so doing so should be preferred and is save to do as well.

Reply:
Thanks, I didn't notice that this was included in `git-repository`.